docs/nodes/maintain/understand-how-a-node-bootstraps.md

```diff
@@ -0,0 +1,197 @@
+its blocks. These blocks would end up poisoning our node local state and making
+it impossible for the node to properly validate the network and reach consensus
+with other nodes.
+
+What is the most reliable source of information in the Avalanche ecosystem? It's
+a *large enough* majority of validators! So the first step of bootstrapping is
+finding enough validators to download containers from.
+
+The P-chain continuously keeps track of validators. So whenever any chain *other
+than the P-chain* has to bootstrap, the P-chain should be able to provide an
+up-to-date list of validators for that Subnet. The node can then reach these
+validators out to securely download containers.
+
+There is a caveat here: the validators list must be *up-to-date*. If the
+validator list is not up-to-date, the node may mistakenly assume that some nodes
+are still validating while their validation period has expired already. This
+would open up the possibility to download faulty blocks from an source that is
+not secure (anymore).
+
+**So every avalanche node must fully bootstrap the P-chain before moving on to
+the other Primary Network chains and other Subnets**.
+
+What about the P-chain? The P-chain can never have an up-to-date validators list
+before completing its bootstrap. To solve this chicken-and-egg situation the
+Avalanche Foundation maintains a trusted set of validators called beacons.
+Beacons Node-IDs and IP addresses are listed in [the AvalancheGo
+codebase](https://github.com/ava-labs/avalanchego/blob/master/genesis/beacons.go).
+Every node has the beacons list available from the start and reach them out as
+soon as they start. 
+
+Beacons and validators are the only trusted sources of information for chains
+content. Beacons and validators availability is key to the bootstrapping process
+so much that **bootstrap stalls until the node establishes secure connections to
+enough beacons or validators**. If the node fails to reach beacons within a
+given timeout, it shuts down as no operation can be carried out securely.
+
+## The Bootstrap Mechanics
+
+Once we understand how to find and connect to validators and beacons, we can
+look at how the block download works.
+
+Let's start by dispelling a common belief: *Avalanche chains are not
+bootstrapped from the genesis up to their frontier*.
+
+Instead, blocks are downloaded from the frontier down to genesis and then
+executed upwards. The frontier is the last accepted block for linear chains and
+the last accepted vertexes for DAGs. 
+
+Why can't we simply download blocks in order, from the genesis upward? The
+reason is again safety: if we downloaded containers upward we wouldn't have a
+way to readily verify whether they are legit or they are fed to our node by some
+malicious peer. Instead by retrieving the frontier securely from a majority of
+honest nodes first, we can cheaply spot any made up container by simply looking
```

Comment:
Genesis is actually never requested since is hardcoded in every node. Also containers are chained, meaning that every child container has the hash or ID of its parent (or parents in the DAG).
What I was trying to say is that if we downloaded containers upwards, we'd have to ask of every single one is valid to a majority of honest nodes, right? I'd be pretty expensive.
Instead we just ask to the majority to validate the frontier. Once frontier is secured we can download all missing containers an just rely on hash checking to verify if containers are safe or not.
Not sure if it's any clearer now. If so, I can try to refurmulate

Reply:
Ah, ok, now I get it. You can request the genesis, but you would then have to verify each child as well. But if you verify the frontier, you have a already agreed upon parent container. And then each parent in the sequence.

Yeah, the explanation could be a bit clearer on that particular point.